### ISSUE TITLE
.net9 RC2 - Compiled Bindings - cannot resolve type in StaticResource

### ISSUE THREAD

**Issue by bcaceiro** [dotnet/maui#25172]
### Description

Using .net9 RC2, I started getting errors in my custom pages setting the page BackgroundColor = {StaticResource XXXX}

Either I am missing something, or there is a new issue with the new compiled bindings, would love some clarification. I managed to hook up a small repro

### Steps to Reproduce

1. Clone my repro and open CompiledBindingsSample https://github.com/bcaceiro/net9-repros
2. Run
3. See error - Cannot resolve type "clr-namespace:CompiledBindingsSample:local:CoreContentPage".

### Link to public reproduction project repository

https://github.com/bcaceiro/net9-repros

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

9.0.0-preview.7.24407.4

### Affected platforms

Android, I was *not* able test on other platforms

### Relevant log output

Cannot resolve type "clr-namespace:CompiledBindingsSample:local:CoreContentPage".

**Comment by bcaceiro** [comment]
@simonrozsival sorry to ping you, but this is the other compile error that I am facing, after upgrading to RC2 ( the same code in RC1 did not produce such error). 

Any guidance how to solve this or is there a issue?

P.S can't select RC2 for the version with problems in the new issue template

**Comment by simonrozsival** [comment]
This looks related to #20917:

```xml
<local:CoreContentPage x:Class="CompiledBindingsSample.MainPage"
                       xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
                       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
                       xmlns:local="clr-namespace:CompiledBindingsSample"
                       x:DataType="local:BaseViewModel"
                       x:TypeArguments="local:BaseViewModel"
                       BackgroundColor="{StaticResource Gray100}">
...
</local:CoreContentPage>
```

The build fails when there's a "non-compilable" `{StaticResource ...}` on a generic type. It looks like we're losing the generic type information somewhere in `Microsoft.Maui.Controls.Build.Tasks.StaticResourceExtensions.FallBack`:

![Image](https://github.com/user-attachments/assets/42ea0b0f-c525-4150-8a51-6c5ba962613c)

/cc @StephaneDelcroix

**Comment by ninachen03** [comment]
This issue has been verified using Visual Studio 17.12.0 Preview 3(9.0.0-rc.2.24503.2). Can repro this issue at Android platform works fine on 17.12.0 Preview 3(9.0.0-preview.7.24407.4)
![Image](https://github.com/user-attachments/assets/95a478b9-1606-4436-98e4-2df643989364)

**Comment by simonrozsival** [comment]
The fix for this issue has been merged into the GA branch: #25287
